ext/RMagick/rmutil.c

```diff
@@ -1557,6 +1703,7 @@ rm_split(Image *image)
 }
 
 
+#if !defined(IMAGEMAGICK_7)
```

Comment:
Could we change the implementation of `rm_check_image_exception` for `IMAGEMAGICK_7` to call `CHECK_EXCEPTION()`? I think that would reduce a lot of the `#if defined(IMAGEMAGICK_7)` calls elsewhere. Then when we are eventually able to remove IM6 support we can replace it in-line.

Reply:
The `rm_check_image_exception` use the `exception` property of the `image` class. This has been fixed in ImageMagick 7 and all methods now get an extra argument called exception. I don't think we can make something common out of this. And as you can see below the exception also needs to be allocated and destroyed. In ImageMagick 6 `new_image->exception` will be used instead.

```c
#if defined(IMAGEMAGICK_7)
    exception = AcquireExceptionInfo();
    (void) SeparateImage(new_image, channel, exception);
    rm_check_exception(exception, new_image, DestroyOnError);
    (void) DestroyExceptionInfo(exception);
#else
    (void) SeparateImageChannel(new_image, channel);
    rm_check_image_exception(new_image, DestroyOnError);
#endif
```